File: api/models/Transaction.js
import { model, Schema } from 'mongoose';
import { v4 as uuidv4 } from 'uuid';

const transactionSchema = new Schema({
  _id: { type: String, default: uuidv4 },
  accountId: { type: String, ref: 'Account' },
  amount: { type: Number, required: true },
  currency: { type: String, enum: ['USD', 'EUR', 'GBP'] },
  type: { type: String, enum: ['deposit', 'withdrawal'] },
  createdAt: { type: Date, default: Date.now },
  updatedAt: { type: Date, default: Date.now }
});

transactionSchema.virtual('account', {
  ref: 'Account',
  localField: 'accountId',
  foreignField: '_id'
});

export default model('Transaction', transactionSchema);

File: nexus_banking_networks/nexus_bank/transaction.py
class Transaction:

    def __init__(self, transaction_id, account_id, amount, attributes):
        self.transaction_id = transaction_id
        self.account_id = account_id
        self.amount = amount
        self.attributes = attributes

    def execute(self, account):
        if self.amount > 0:
            account.deposit(self.amount)
        else:
            account.withdraw(-self.amount)
